### PR TITLE
[Cross Repo Search] Namespace detail, collection list and collection detail refactor

### PR DESCRIPTION
**Pull request by jerabekjiri**
Issue: AAH-767

Requires:
- https://github.com/ansible/galaxy_ng/pull/1649 
- ~https://github.com/ansible/galaxy_ng/pull/1642~
- ~https://github.com/ansible/galaxy_ng/pull/1658~
- ~https://github.com/pulp/pulp_ansible/pull/1395~
- ~https://github.com/ansible/galaxy_ng/pull/1652~
- ~https://github.com/pulp/pulp_ansible/pull/1357~

Use new cross repo search for.
- Namespace detail
- Collection list (search)
- Collection detail (install, documentation, contents, import log, dependencies)

Changes: 
- new distributions tab in collection detail Screenshot from 2023-03-28 00-49-02
Screenshot from 2023-03-27 12-19-28
- removed repo selector for namespace detail
- changed URL for namespace detail to /ui/namespaces/<namespace>
- changed URL for collection list (search.tsx) to /ui/collections
- repository badge in collection card and list item
Screenshot from 2023-03-27 11-58-03
Screenshot from 2023-03-27 11-37-14

### CODE REVIEW

File: src/containers/namespace-detail/namespace-detail.tsx [hunk 1]
```diff
@@ -437,7 +422,6 @@ export class NamespaceDetail extends React.Component<IProps, IState> {
                   collections={collections}
                   itemCount={itemCount}
                   showControls={this.state.showControls}
-                  repo={this.context.selectedRepo}
```
Comment (newswangerd) [inline src/containers/namespace-detail/namespace-detail.tsx:440]: Show the namespace logo on the collection list view (like the list layout on the search page does now)

File: src/components/collection-detail/collection-info.tsx [hunk 2]
```diff
@@ -76,67 +88,97 @@ export const CollectionInfo = ({
...
+                    To download this collection, configure your client to
+                    connect to one of this repositories distributions.{' '}
+                    <Link
+                      to={formatPath(Paths.collectionDistributionsByRepo, {
+                        repo: repository.name,
+                        namespace: collection_version.namespace,
+                        collection: collection_version.name,
+                      })}
+                    >
+                      <ExternalLinkAltIcon />
```
Comment (newswangerd) [inline src/components/collection-detail/collection-info.tsx:124]: put the link on the "distributions" word. The `ExternalLinkAltIcon` should only be used for links to external web pages.

**Review by newswangerd — CHANGES_REQUESTED** [review]
Mostly looking pretty good. Found a few UX things and some bugs

File: src/containers/search/search.tsx [hunk 3]
```diff
@@ -144,7 +146,13 @@ class Search extends React.Component<RouteProps, IState> {
     } = this.state;
     const noData =
       collections.length === 0 &&
-      !filterIsSet(params, ['keywords', 'tags', 'sign_state']);
+      !filterIsSet(params, [
+        'keywords',
+        'tags',
+        'is_signed',
+        'repository__name',
```
[the remark this answers is not in the document]
Comment (jerabekjiri) [reply src/containers/search/search.tsx:153]: nice catch

File: src/containers/search/search.tsx [hunk 4]
```diff
@@ -144,7 +146,13 @@ class Search extends React.Component<RouteProps, IState> {
     } = this.state;
     const noData =
       collections.length === 0 &&
-      !filterIsSet(params, ['keywords', 'tags', 'sign_state']);
+      !filterIsSet(params, [
```
[the remark this answers is not in the document]
Comment (jerabekjiri) [reply src/containers/search/search.tsx:149]: fixed, now it only sends requests if repository filter is selected

**Review by newswangerd — APPROVED** [review]
Overall looks good. Only thing I noticed is unpriviledged users can see the control dropdown on the list view. We don't have a good solution for this right now, so I would say just check the user's global permissions to show this.

Same problem on the collection page. For this one you can query the namespaces API to get the user's permissions on the given namespace.

image

**Comment by himdel** [comment]
Screenshot test failure expected - removes the repo filter from collections :+1: 
(Insights are also unrelated)
restarting collection_upload...

**Comment by jerabekjiri** [comment]
@himdel I'm trying to figure out what is the issue with collection_upload, It's passing locally

**Pull request merged**